script/BaseScript.sol

```diff
@@ -0,0 +1,182 @@
+
+import {Script, stdJson} from "lib/forge-std/src/Script.sol";
+
+import {IProvider} from "src/interface/IProvider.sol";
+import {TestnetActors, IActors, MainnetActors} from "script/Actors.sol";
+import {BscContracts, ChapelContracts, IContracts} from "script/Contracts.sol";
+import {VaultUtils} from "script/VaultUtils.sol";
+
+import {IVaultViewer} from "src/interface/IVaultViewer.sol";
+import {BaseVaultViewer} from "src/utils/BaseVaultViewer.sol";
+import {Vault} from "src/Vault.sol";
+
+import {TransparentUpgradeableProxy as TUP} from
+    "lib/openzeppelin-contracts/contracts/proxy/transparent/TransparentUpgradeableProxy.sol";
+
+import {TimelockController} from "lib/openzeppelin-contracts/contracts/governance/TimelockController.sol";
+import {Strings} from "lib/openzeppelin-contracts/contracts/utils/Strings.sol";
+
+import {ProxyUtils} from "script/ProxyUtils.sol";
+
+abstract contract BaseScript is Script, VaultUtils, ProxyUtils {
+    using stdJson for string;
+
+    uint256 public minDelay;
+    IActors public actors;
+    IContracts public contracts;
+
+    address public deployer;
+    TimelockController public timelock;
+    IProvider public rateProvider;
+    Vault public vault;
+    Vault public implementation;
+    IVaultViewer public viewer;
+    IVaultViewer public viewerImplementation;
+
+    error UnsupportedChain();
+    error InvalidSetup();
+
+    // needs to be overriden by child script
+    function symbol() public view virtual returns (string memory);
+
+    function _setup() public virtual {
+        deployer = msg.sender;
+
+        if (block.chainid == 97) {
+            minDelay = 10 seconds;
+            TestnetActors _actors = new TestnetActors();
+            actors = IActors(_actors);
+            contracts = IContracts(new ChapelContracts());
+        }
+
+        if (block.chainid == 56) {
+            minDelay = 1 days;
+            MainnetActors _actors = new MainnetActors();
+            actors = IActors(_actors);
+            contracts = IContracts(new BscContracts());
+        }
+    }
+
+    function _verifySetup() public view virtual {
+        if (block.chainid != 56 && block.chainid != 97) {
+            revert UnsupportedChain();
+        }
+        if (
+            address(actors) == address(0) || address(contracts) == address(0) || address(rateProvider) == address(0)
+                || address(timelock) == address(0)
+        ) {
+            revert InvalidSetup();
+        }
+    }
+
+    function _deployViewer(address viewerImplementation_) internal virtual {
+        if (address(vault) == address(0) || address(viewerImplementation_) == address(0)) {
+            revert InvalidSetup();
+        }
+
+        viewerImplementation = IVaultViewer(payable(viewerImplementation_));
+
+        bytes memory initData = abi.encodeWithSelector(BaseVaultViewer.initialize.selector, address(vault));
+
+        TUP proxy = new TUP(address(viewerImplementation), actors.ADMIN(), initData);
```

Comment:
we use the full name on the other side and here this shortened version let's use the longer version everywhere for clarity

Reply:
I'm not sure what you mean here.  it looks identical to the kernel-lrt BaseScript.  did you want to use another word besides viewer or proxy?